scipy/signal/ltisys.py

```diff
@@ -2494,3 +2497,271 @@ def place_poles(A, B, poles, method="YT", rtol=1e-3, maxiter=30):
+        \begin{array}{lcl} P A P^T &=& \bar A \\
+                            P B &=& \bar B\end{array}
+
+    The new system has the following form:
+
+    .. math::
+
+        \bar A = \left[ \begin{array}{cc}
+                        H_c & H_{xx} \\
+                        0 & H_{nc}
+                        \end{array} \right]
+
+    :math:`\bar A` is a block upper Hessenberg matrix hence the `H` prefix used
+    for its blocks.
+
+    .. math::
+
+        \bar B = \left [ \begin{array}{c} B_c \\ 0 \end{array} \right ]
+
+
+    :math:`H_c, B_c` is the controllable part of the original system. If the
+    real part of the eigenvalues of :math:`H_{nc}` are negative the system
+    will be stabilizable.
+
+    Parameters
+    ----------
+    A, B : ndarray
+        State-space representation of linear system :math:`AX+BU`.
+
+    C : ndarray, optionnal
+        The output matrix of the linear system (i.e :math:`Y=CX`).
+        Default is None.
+
+    Returns
+    -------
+    hessenberg_form : Bunch object
+        hessenberg_form is composed of:
+            Abar, Bbar : ndarray
+                The controller Hessenberg form of the linear system
+                :math:`AX+BU`
+
+            Cbar : ndarray
+                Optionnaly if C (output matrix) is given the function will also
+                return :math:`\bar C = C P` otherwise `Cbar = None` will be
+                returned. Cbar has the following form:
+
+                 .. math::
+                     \bar C = \left[ \begin{array}{cc}
+                                     C_c & C_{nc}
+                                     \end{array} \right]
+
+                With :math:`C_c` such that the transfer function of (A,B,C) and
+                (Hc,Bc,Cc) are the same.
+
+            P : ndarray
+                The orthogonnal similarity transformation
+
+            control_index : int
+                The controllability index, i.e the number of controllable
+                states, of the linear system :math:`AX+BU`. If
+                ``control_index == A.shape[0]`` then the linear system is
+                controllable.
+
+    Examples
+    --------
+    A simple example demonstrating how to check the control index of a linear
+    system.
+
+    >>> from scipy import signal
+
+    >>> A = np.array([[ 1.380,  -0.2077,  6.715, -5.676  ],
+    ...               [-0.5814, -4.290,   0,      0.6750 ],
+    ...               [ 1.067,   4.273,  -6.654,  5.893  ],
+    ...               [ 0.0480,  4.273,   1.343, -2.104  ]])
+    >>> B = np.array([[ 0,      5.679 ],
+    ...               [ 1.136,  1.136 ],
+    ...               [ 0,      0,    ],
+    ...               [-3.146,  0     ]])
+    >>> hessenberg_form = signal.controller_hessenberg(A, B)
+    >>> hessenberg_form.control_index
+    4
+
+    References
+    ----------
+    .. [1] Datta Biswa, "Numerical Methods for Linear Control Systems",
+           Elsevier, pp 173-183, 2003.
+
+    """
+
+    if A.ndim != 2:
```

Comment:
You should probably precede these with `A = np.asarray(A)` and `B = np.asarray(B)` to ensure you actually have `.ndim` properties available.